### PR TITLE
🔨 Allow Mergify to detect more failing cases for CI restarts & Automatically add PRs to queue

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

In our CI pipeline there are tests that fail but will still make the `unit-tests`, `integration-tests` and `system-tests` be green. This did not allow Mergify's Workflow Automation to restart the CI.
Added all the edge cases that could make the restart not occur.

Instead of manually enqueuing the PRs, Mergify will now add them automatically once conditions are met.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
thx

MINOR, the correct icon is  ` 🔨    Add or update development scripts.`. and tag it as `t:maintenance` so we can account for it in the stats

**Review by sanderegg — APPROVED** [review]
you are re-writing the github workflow file.. is this not contraproductive?

File: .github/mergify.yml [hunk 1]
```diff
@@ -16,17 +13,17 @@ queue_rules:
       - "#changes-requested-reviews-by=0" # No changes requested
       - "#review-threads-unresolved=0" # All review threads resolved
 
-    merge_conditions:
+    merge_conditions: # conditions to be met before being able to merge
       # list of CI checks that need to pass
+      - check-success=system-test-environment-setup
+      - check-success=changes
+      - check-success=build-test-images
+      - check-success=build-test-images-frontend
```
[the remark this answers is not in the document]
Comment (GitHK) [reply .github/mergify.yml:21]: I need to check if these can be removed. For now, these work as well. I'm not done figuring out every aspect of the merge queue.
